src/accounts/accountLoginMiddleware.ts

```
@@ -0,0 +1,65 @@
+import {Middleware} from 'polka';
+import send from '@polka/send-type';
+
+import {isEmail, normalizeEmail} from '../client/email/email.js';
+import {Server} from '../server/Server.js';
+import {AccountModel} from './AccountModel.js';
+import {ClientSession} from '../client/session/session.js';
+
+export const accountLoginMiddleware = (server: Server): Middleware => {
+	return async (req, res) => {
+		console.log('accountLoginMiddleware req.body', req.body!.email); // TODO logging
+
+		// TODO formalize and automate validation and normalization
+		const email = normalizeEmail(req.body!.email);
+		if (!isEmail(email)) {
+			return send(res, 400, {reason: `Invalid email '${email}'.`});
+		}
+
+		if (req.account) {
+			if (req.account.email === email) {
+				return send(res, 400, {reason: `Already logged in with this email.`});
+			} else {
+				// TODO what's the desired behavior here?
+				// We probably want to log the user out and back in with this new email.
+				// However we need to be careful that the session and account get updated properly.
+				// For now just return an error - the client UI should not allow this in any case.
```

I'd say if a user is already logged in we should just return a single error, instead of trying to get fancy. i.e. "You can't log in because you are already logged in. Log out if you wish to log in with a different account."